packages/lodestar/src/metrics/server/http.ts

```diff
@@ -24,8 +24,13 @@ export class HttpMetricsServer implements IMetricsServer {
   public async start(): Promise<void> {
     if (this.opts.enabled) {
       this.logger.info(`Starting metrics HTTP server on port ${this.opts.serverPort}`);
-      // @ts-ignore
-      await promisify<void, number>(this.http.listen.bind(this.http))(this.opts.serverPort);
```

Comment:
Can you remember why ts-ignore was needed with promisify impl?

Also it would probably be nice to catch error here if port is taken^^

Reply:
Because Typescript is not able to figure out the complex overloads of the listen function and results in a bad type. So `listen(3000)` results in a type error.

Reply:
Oh yeah, that was so annoying, it was same thing in rpc server (when we had it).

Reply:
I'm still confused in what's the appropriate way of doing a listen handler. Does the callback return an error? According to the official types it doesn't.

Reply:
Not super familiar, but it could be that error is first param in callback

Reply:
Seems that the callback is subscribed to the `'listening'` event https://github.com/nodejs/node/blob/f63436d190b60e12131036aa9d1888d9023e9127/lib/net.js#L1378, which is emitted without any arguments.